src/System.Runtime.WindowsRuntime/tests/System/IO/CreateSafeFileHandleTests.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using Windows.Storage;
+using Xunit;
+
+namespace System.IO
+{
+    public class CreateSafeFileHandleTests
+    {
+        [Fact]
+        public void NullStorageFile_ThrowsArgumentNull()
+        {
+            IStorageFile file = null;
+            Assert.Equal(
+                "windowsRuntimeFile",
+                Assert.Throws<ArgumentNullException>(() => file.CreateSafeFileHandle()).ParamName);
+        }
+
+        [Fact]
+        public void FromStorageFile_BadAccessThrowsOutOfRange()
+        {
+            IStorageFile file = new StorageFileMock();
+            Assert.Equal(
+                "access",
+                Assert.Throws<ArgumentOutOfRangeException>(() => file.CreateSafeFileHandle((FileAccess)100)).ParamName);
+        }
+
+        [Fact]
+        public void FromStorageFile_BadSharingThrowsOutOfRange()
+        {
+            IStorageFile file = new StorageFileMock();
+            Assert.Equal(
+                "share",
+                Assert.Throws<ArgumentOutOfRangeException>(() => file.CreateSafeFileHandle(FileAccess.ReadWrite, (FileShare)100)).ParamName);
```

Comment:
should also check a bad value on the other side of the range e.g. -1 for completeness.

Same comment for the below tests expecting an AOORE